src/test/archive/archive_node_tests/archive_node_tests.ml

```diff
@@ -27,6 +27,86 @@ let assert_archived_blocks ~archive_uri ~expected =
       actual_blocks_count expected ()
   else Deferred.unit
 
+(* Convert performance metrics to a JSON format suitable for output *)
+(* The metrics are expected to be a list of tuples (operation, avg_time) *)
+(* where operation is a string and avg_time is a float representing the average time in milliseconds *)
+let perf_metrics_to_yojson metrics =
+  let json_list =
+    List.map metrics ~f:(fun (operation, avg_time) ->
+        `Assoc
+          [ ("operation", `String operation); ("avg_time_ms", `Float avg_time) ] )
+  in
+  `List json_list
+
+(* Extract performance metrics from the log file *)
+(* where X is a floating point number representing the time taken for the operation *)
+(* log output should be in JSON format *)
+(* Example log line: {..., "message": "Operation took 123.45 ms"} *)
+(* The function will return a map of operation names to their average time in milliseconds *)
+let extract_perf_metrics log_file =
+  let open Deferred.Let_syntax in
+  let%bind lines = Reader.file_lines log_file in
+  let perf_metrics =
+    List.filter_map lines ~f:(fun line ->
+        match Logger.Message.of_yojson (Yojson.Safe.from_string line) with
+        | Ok entry ->
+            if String.is_substring entry.message ~substring:" took " then
+              (* Extract the operation and time from the line *)
+              (* Parse the JSON line to extract the message field *)
+              let pattern =
+                Re.Perl.compile_pat {|(.+) took (\d+(?:\.\d+)?)(ms|us)|}
```

Comment:
I think using regex to parse a string is anti-pattern, when we could have just store the data in a structural way.